src/handle.rs

```diff
@@ -212,6 +212,10 @@ pub struct Global<T> {
   isolate_handle: IsolateHandle,
 }
 
+// Globals can go between threads as long as a locker was used to acquire a SharedIsolate.
+unsafe impl<T> Send for Global<T> {}
+unsafe impl<T> Sync for Global<T> {}
```

Comment:
In @Earthmark 's previous PR #1411 , there is a comment saying, simply add `Send` and `Sync` trait to `Global` can be risk.

Could you also please give comments?

Reply:
I found there is some mutex on the `isolate` here: https://github.com/denoland/rusty_v8/blob/b04640d3f6f5a295d2cabd8fd72b86d02ae1b7d0/src/isolate.rs#L1490

This code looks it wants to use a mutex to lock the isolate, makes me feel like using the `Locker` C++ api is the correct way to implement this part. Once this done, seems a `Isolate` can natively support `Send` and `Sync` traits, i.e. safely be sent between threads.

Correct?

Reply:
I completely forgot I had this change out here, thank you for picking it back up. 

Let me know if you have questions about what I was trying to do.

I'm pretty sure the comment on the original pr about globals stands. 

Globals moving between threads is a bit scary if lockers are not used, although it's kind of a landmine either way because the global needs to be made into a local with the same isolate that made it. 

If lockers are not used, the global can only be used on the isolate that made it, which will be single threaded.

This usage constraint kind of enforces that the global is single threaded, but a global being send means it could be moved to another thread accidently (like if it's bundled in an async method), then dropped, which I think I vaguely recall could be a problem. (I haven't looked at this in months, and I'm writing this on my phone).

I think there are two paths with globals: 

verify that dropping a global on the wrong thread won't explode (even in the lockers-not-used case)

OR

Make some other thread safe wrapper for globals that locked isolates can produce, that has whatever mechanisms are needed to not explode.

That being said, haven't looked at this in months.


I'm not sure if it's acceptable to make every isolate internally use lockers, they are re-entrant, but adding lockers to every API may add an unacceptable performance cost for users who are not using lockers.

I'm pretty sure lockers block as well, which is a bit nasty if someone wasn't expecting a hard block from an API call.

Reply:
Agree, too many API are designed to run in a single thread at the beginning. And JavaScript language itself doesn't support such concepts like multi-thread and mutex.

So just keep it `!Send` and `!Sync` is more compatible with V8 and JavaScript language.

I close this PR now.